test/cypress/integration/group-paying.spec.js

```diff
@@ -42,47 +52,35 @@ describe('Group Payments', () => {
       invitationLinks.anyone = url
     })
 
-    addIncome(true, 250)
+    setIncomeDetails(true, 250)
     cy.giLogout()
   })
 
   it('Three users join the group and add their income details', () => {
-    cy.giAcceptGroupInvite(invitationLinks.anyone, {
-      username: `user2-${userId}`,
-      groupName,
-      bypassUI: true,
-      actionBeforeLogout: () => {
-        addIncome(false, 900)
-      }
-    })
+    const options = { groupName, bypassUI: true, shouldLogoutAfter: false }
+    cy.giAcceptGroupInvite(invitationLinks.anyone, { username: `user2-${userId}`, ...options })
+    setIncomeDetails(false, 900)
+    cy.giLogout()
 
-    cy.giAcceptGroupInvite(invitationLinks.anyone, {
-      username: `user3-${userId}`,
-      groupName,
-      bypassUI: true,
-      actionBeforeLogout: () => {
-        addIncome(false, 750)
-      }
-    })
+    cy.giAcceptGroupInvite(invitationLinks.anyone, { username: `user3-${userId}`, ...options })
+    setIncomeDetails(false, 750)
+    cy.giLogout()
 
-    cy.giAcceptGroupInvite(invitationLinks.anyone, {
-      username: `user4-${userId}`,
-      groupName,
-      bypassUI: true,
-      shouldLogoutAfter: false,
-      actionBeforeLogout: () => {
-        addIncome(true, 100)
-      }
-    })
+    cy.giAcceptGroupInvite(invitationLinks.anyone, { username: `user4-${userId}`, ...options })
+    setIncomeDetails(true, 100)
   })
 
-  it('user1 sends to user2 $71.43 (total)', () => {
+  it('user1 sends $71.43 to user2 (total)', () => {
     cy.giSwitchUser(`user1-${userId}`, { bypassUI: true })
 
     cy.getByDT('paymentsLink').click()
     cy.get('[data-test-date]').should('have.attr', 'data-test-date', humanDateToday)
 
-    assertPaymentsTabs(['Todo2', 'Sent'])
+    assertNavTabs(['Todo2', 'Sent'])
+    assertMonthOverview([
+      'Payments sent0 out of 2',
```

Comment:
Even though the HTML gets squished down into this text string, is it possible to separate these into arrays of strings to look for within the `li` in the assertion on line 38? Then it might be easier to read and understand as we add, update, satisfy, and fix tests.

Reply:
I remember trying something similar a few months ago but couldn't make it work (The same problem happens with Testing Library too). If you find a solution please @ me so that I can learn it too :D